packages/legend-studio/src/stores/sidebar-state/LocalChangesState.ts

```diff
@@ -559,8 +559,12 @@ export class LocalChangesState {
             entityChanges: localChanges,
             revisionId: this.sdlcState.activeRevision.id,
           },
-        )) as PlainObject<Revision>,
+        )) as unknown as PlainObject<Revision> | undefined;
+      const revisionChange = guaranteeNonNullable(
+        nullableRevisionChange,
+        `Can't push empty change set. This may be due to an error with your local change set. Please considering raising issue with developer.`,
```

Comment:
I don't think we need to be lengthy and mention the developer part. You just need to wrap `IllegalStateError` and `notifyError` would handle it. Also it's not really an error with local changeset but more like change detection

```suggestion
        `Can't push an empty change set. This may be due to an error with change detection`,
```

I'm not sure what to tell users here to avoid `TMI` and to tell them that they are in someway alright 😅